Define a land cover resource that can be used for biodiversity and biocontrol zones

cookeac opened the pull request:
LandCoverResource has:
* id, identifiers, name, links, resourceType from `ResourceType`
* totalLength/totalArea from `SpatialResourceType`
* arrays of classifications and activities from `LandUseResourceType`
* `landCoverClass` - a structured categorisation using a scheme that defines the type of land cover (e.g. NZ, AU, or UN land covers)
* `speciesPresent` - species that may be present with relative abundance and ecological group
* `eligibility` - eligibility under some government or other classification

Resolves #22 

**Note: You won't be able to merge this successfully until #28 is merged (depends on this for base classes).**

cookeac committed:
Define LandCoverResource and types it uses

cookeac committed:
Create DAFORabundance.json

DAFOR ecology species abundance scale

VicXue commented on types/ComplianceEligibilityType.json:
```diff
@@ -0,0 +1,19 @@
+    
+    "allOf": [
+        {
+            "$ref": "../types/ClassificationType.json"
```
Do we need this reference both here and at the [top level of the LandCoverResource too (line 13)](https://github.com/Datalinker-Org/Geospatial/pull/35/files#diff-ae657d3d803e0ae1872faeb90afdd1d6cdc7fb8659d529f2b3f690f38c5e379dR13)?

VicXue replied:
There are three occurrences of the ClassificationType in [LandUseResourceType](https://github.com/Datalinker-Org/Geospatial/pull/35/files#diff-ae657d3d803e0ae1872faeb90afdd1d6cdc7fb8659d529f2b3f690f38c5e379dR5), and those two mentioned above. Is it meaningful to keep all three of them?

cookeac replied:
I think its meaningful, but perhaps duplicative. For instance, it would be possible to represent `landCoverClass` and `eligibility` within the existing `classifications[]` introduced in `LandUseResourceType` (the "parent" class that `LandCoverResource` extends).

However, it's not readily discoverable - someone would need to know to work through the `classifications[]` array looking for these types. In particular, `landCoverClass` is a structured method of defining the type of land cover we are talking about (for instance, starting with forest and then indigenous forest), which is the key thing we need to know about LandCoverResources. So I think while there is some duplication, it is better to break it out for clarity.

VicXue commented on types/ComplianceEligibilityType.json:
```diff
@@ -0,0 +1,19 @@
+
+            "properties": {
+                "eligible": {
+                    "type": "boolean",
```
In PF, this field is a string instead of a boolean.

cookeac replied:
That's a good point. I had thought about True/False but its possible there are other eligibility values and a simple string might be more flexible.

cookeac committed:
Update to land cover resource and eligibility

Improved the description of a land cover resource. Changed the type of "eligible" from a boolean to a string.
